tests/syscall/fs/host/host.c

```diff
@@ -78,26 +89,105 @@ int wmain(int argc, wchar_t* argv[])
     // Windows does not support umask.
     // Please set up the right permission to the parent directory.
 
-    int ret = test_fs_posix(enclave_path, src_dir, tmp_dir);
+    rmdir(argv[3]);
 
-    free(src_dir);
-    free(tmp_dir);
+    r = oe_create_fs_enclave(enclave_path, type, flags, NULL, 0, &enclave);
+    OE_TEST(r == OE_OK);
 
-    return ret;
+    r = test_fs(enclave, src_dir, tmp_dir);
+    OE_TEST(r == OE_OK);
+
+    r = oe_terminate_enclave(enclave);
+    OE_TEST(r == OE_OK);
+
+    printf("=== passed all tests (hostfs)\n");
+
+    return 0;
 }
+#else
+int rmdir(const char* path)
+{
+    char abs_path[PATH_MAX];
+    int result = -1;
+    struct dirent* entry = NULL;
+    DIR* dir = NULL;
+    if (!(dir = opendir(path)))
+    {
+        goto done;
+    }
+
+    while ((entry = readdir(dir)))
```

Comment:
Non-blocking:

@jhand2 Can you advise on an idiomatic way to do this in linux? A couple of suggestions based on my limited exposure:

* Should we be using `nftw()` here to skip navigating links during the directory walk?
* Is it a better practice to use `stat()` and `S_ISREG`/`S_ISDIR` to figure out the type of the item, instead of opening and closing it?

Reply:
Yes nftw is a better tool to avoid trivial file operations.

Reply:
I would use nftw and then call the remove syscall (https://linux.die.net/man/3/remove) in the callback which will call `unlink` for files and `rmdir` for directories.

nftw does a DFS when passed `FTW_DEPTH` so by the time you remove the directory it should be empty.

This is untested but it would be something like:

```
int nftw_callback(const char *path, const struct stat *sb, int flags, struct FTW *ftwbuf)
{
    int ret = remove(path);
    if (ret != 0)
        fprintf(stderr, "Could not remove file %s\n", path);

    return ret;
}

int rmdir_recusrive(char *path)
{
    // FTW_PHYS won't follow symlinks. This is mostly because following symlinks is
    // scary and you could end up destroying the filesystem
    return nftw(path, nftw_callback, 64 /* depth limit */, FTW_DEPTH | FTW_PHYS);
}
```

Reply:
Fixed, thanks!